inc/functions_search.php

```diff
@@ -150,6 +150,82 @@ function get_unsearchable_forums($pid=0, $first=1)
 	return $unsearchable;
 }
 
+/**
+ * Build query condition for threads/posts the user is allowed to see.
+ * Will return for example:
+ *  - visible = 1 - for normal users
+ *  - visible >=1 - for admins & super mods
+ *  - (visible = 1 OR (visible = ? AND fid IN ...)) - for forum moderators
+ * 
+ * @param string $table_alias The alias of the table eg t to use t.visible instead of visible
+ * @return string the query condition 
+ */
+function get_visible_where($table_alias = null)
+{
+	global $db, $mybb;
+
+	$aliasdot = '';
+	if(!empty($table_alias))
+	{
+		$aliasdot = $table_alias.'.';
+	}
+
```

Comment:
Although I'm not sure if I'm right but following https://github.com/mybb/mybb/pull/4091#discussion_r445985317 , maybe we should change here into
```suggestion

	// Common for all users.
	$unapproved_where = "{$aliasdot}visible = 1";
	if($mybb->user['uid'] && $mybb->settings['showownunapproved'])
	{
		// Registered user should be able to see their own unapproved threads/posts but not guests, if the global setting permits.
		$unapproved_where .= " OR ({$aliasdot}visible = 0 AND {$aliasdot}uid = '{$mybb->user['uid']}')";
	}
```